### PR TITLE
feat: Add _distributeRelayerRefund unit tests

### PR DESCRIPTION
**Pull request by nicholaspai**
Also refactors the SpokePool.sol code so that the validation of merkle leaf properties are done as close to their usage within the logic as possible

### CODE REVIEW

File: contracts/SpokePool.sol [hunk 1]
```diff
@@ -1376,15 +1370,6 @@ abstract contract SpokePool is
     // Should be overriden by implementing contract depending on how L2 handles sending tokens to L1.
     function _bridgeTokensToHubPool(uint256 amountToReturn, address l2TokenAddress) internal virtual;
 
-    function _validateRelayerRefundLeaf(
-        uint256 _chainId,
-        address[] memory refundAddresses,
-        uint256[] memory refundAmounts
-    ) internal view {
-        if (_chainId != chainId()) revert InvalidChainId();
```
Comment (nicholaspai) [inline contracts/SpokePool.sol:1384]: I moved these two checks so that they're closer to the logic that uses the variables checked

Comment (mrice32) [reply contracts/SpokePool.sol:1384]: +1 good for readability

**Review by mrice32 — APPROVED** [review]
LGTM!

**Review by pxrl — APPROVED** [review]
LGTM

**Pull request merged**